scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/RuntimeReducerEstimator.scala

```diff
@@ -43,69 +74,136 @@ trait RuntimeEstimationScheme {
    *  the time that a "typical" reducer took in a "typical" job.
    *  Suggested implementation: mean or median.
    */
-  def estimateJobTime(times: Seq[Long]): Option[Long]
+  def estimateJobTime(times: Seq[Double]): Option[Double]
 }
 
 trait BasicRuntimeReducerEstimator extends HistoryReducerEstimator {
   import RuntimeReducerEstimator._
 
+  private val LOG = LoggerFactory.getLogger(this.getClass)
+
   def runtimeEstimationScheme: RuntimeEstimationScheme
 
   def estimateReducers(info: FlowStrategyInfo, history: Seq[FlowStepHistory]): Option[Int] = {
-    val reduceTimes: Seq[Seq[Long]] = getReduceTimes(history)
+    val reduceTimes: Seq[Seq[Double]] = getReduceTimes(history)
+
+    LOG.info(
+      s"""|
+          |History items have the following numbers of tasks:
+          | ${history.map(_.tasks.length)},
+          |and the following numbers of tasks have valid task histories:
+          | ${reduceTimes.map(_.length)}""".stripMargin)
 
     // total time taken in the step = time per reducer * number of reducers
-    val jobTimes: Seq[Option[Long]] = reduceTimes.map { xs => runtimeEstimationScheme.estimateTaskTime(xs).map(_ * xs.length) }
+    val jobTimes: Seq[Option[Double]] = reduceTimes.map { xs => runtimeEstimationScheme.estimateTaskTime(xs).map(_ * xs.length) }
 
     // time per step, averaged over all the steps
-    val typicalJobTime: Option[Long] = runtimeEstimationScheme.estimateJobTime(jobTimes.flatten)
+    val typicalJobTime: Option[Double] = runtimeEstimationScheme.estimateJobTime(jobTimes.flatten)
 
-    val desiredRuntime = getRuntimePerReducer(info.step.getConfig)
-    typicalJobTime.map { t => (t.toDouble / desiredRuntime).ceil.toInt }
+    val desiredRuntime: Long = getRuntimePerReducer(info.step.getConfig)
+
+    val estimate = typicalJobTime.map { t: Double => (t / desiredRuntime).ceil.toInt }
+
+    LOG.info(s"""
+      | - Job times: $jobTimes
```

Comment:
Just wondering if logging the whole sequence might be too verbose? The other values below are `Option`s so should be fine.